vignettes/tutorial/custom-inputs.rmd

```diff
@@ -0,0 +1,236 @@
+        string_list(values)
+      )
+    })
+    output$button_text <- renderText({
+      req(input$button)
+      values <- input$button
+      paste0(
+        "Button: ",
+        string_list(values)
+      )
+    })
+
+    output$dropdown_text <- renderText({
+      req(input$dropdown)
+      values <- input$dropdown
+      paste0(
+        "Dropdown: ",
+        string_list(values)
+      )
+    })
+
+    output$text <- renderText({
+      req(input$text)
+      values <- input$text
+      paste0(
+        "Dropdown: ",
+        string_list(values)
+      )
+    })
+  }
+)
+
+```
+
+![](images/custom-inputs.gif)
+
+Example of saving the state when changing the page:
+
+```R
+# helper function
+update_table <- function(data, id, column, value, key_column = NULL) {
+  if (!is.null(key_column)) {
+    data[data[[key_column]] == id, column] <- value
+  } else {
+    data[id, column] <- value
+  }
+  return(data)
+}
+
+# in server.R
+values <- input$text
+updateReactable(
+  "react",
+  data = update_table(
+    df,
+    values$row,
+    values$column,
+    values$value
+  )
+)
+```
+
+
+You can also use `{tippy}` to add tooltips to your reactable columns:
+
+```R
+library(shiny)
+library(reactable)
+library(reactable.extras)
+string_list <- function(values) {
+  paste0(
+    "{", paste0(names(values), " : ", unlist(values), collapse = ", "), "}"
+  )
+}
+
+df <- MASS::Cars93[, 1:4]
+df$Date <- sample(seq(as.Date("2020/01/01"),
+                      as.Date("2023/01/01"),
+                      by = "day"),
+                  nrow(df))
+df$Check <- sample(c(TRUE, FALSE), nrow(df), TRUE)
+
+shinyApp(
+  ui = fluidPage(
+    reactable.extras::reactable_extras_dependency(),
+    reactable_extras_ui("table"),
+    hr(),
+    textOutput("date_text"),
+    textOutput("button_text"),
+    textOutput("check_text"),
+    textOutput("dropdown_text"),
+    textOutput("text")
+  ),
+  server = function(input, output) {
+    reactable_extras_server(
+      "table",
+      data = df,
+      columns = list(
+        Manufacturer = colDef(
+          header = tooltip_extra("tooltip", content = "Manufacturer type"), #tooltip
+          cell = button_extra("button", class = "button-extra")
+        ),
+        Check = colDef(
+          header = tooltip_extra("tooltip", content = "Checkbox"),
+          cell = checkbox_extra("check", class = "checkbox-extra"),
+          align = "left"
+        ),
+        Date = colDef(
+          header = tooltip_extra("tooltip", content = "Date input"),
+          cell = date_extra("date", class = "date-extra")
+        ),
+        Type = colDef(
+          header = tooltip_extra("tooltip", content = "Type dropdown"),
+          cell = dropdown_extra(
+            "dropdown",
+            unique(df$Type),
+            class = "dropdown-extra"
+          )
+        ),
+        Model = colDef(
+          header = tooltip_extra("tooltip", content = "Model input"),
+          cell = text_extra(
+            "text"
+          )
```

Comment:
The first argument "tooltip" on the `tooltip_extra` function is not necesarry. We should remove it.

Reply:
Updated! I noticed the background is white, but for general usage I think it's better